Tasks/PublishPipelineArtifactV1/task.json

```diff
@@ -0,0 +1,44 @@
+{
+    "id": "ECDC45F6-832D-4AD9-B52B-EE49E94659BE",
+    "name": "PublishPipelineArtifact",
+    "friendlyName": "Publish pipeline artifact",
+    "description": "Publish a local directory or file as a named artifact for the current pipeline",
+    "helpUrl": "https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/publish-pipeline-artifact",
+    "helpMarkDown": "Publish a local directory or file as a named artifact for the current pipeline.",
+    "category": "Utility",
+    "author": "Microsoft Corporation",
+    "version": {
+        "Major": 1,
+        "Minor": 0,
+        "Patch": 0
+    },
+    "groups": [],
+    "demands": [],
+    "preview": true,
+    "minimumAgentVersion": "2.140.1",
+    "inputs": [
+        {
```

Comment:
Re: https://github.com/microsoft/azure-pipelines-agent/pull/2279#pullrequestreview-244012048 ... if we make `artifactName` optional (since it can be calculated using the job ID) can we re-order the inputs so "path" appears first (since it's always required)? note: I would hope the web UI is smart enough to list required inputs first, but in case it doesn't ...

Reply:
yes. makes sense, it'll be re-ordered on the UI as well. :)